### PR TITLE
Bugfix for parquet metadata writes of empty dataframe partitions (pyarrow) 

### PR DESCRIPTION
**Pull request by callumanoble**
Attempting to write metadata for empty dataframe partitions with `to_parquet` (pyarrow impl.) can raise unintended exceptions or cause segfaults.

This PR filters out `None` meta data elements generated by `write_partition` that cause the issues. Unit tests cases demonstrating the current issue are included.

More details in the following comment https://github.com/dask/dask/issues/6600#issuecomment-710041237

### CODE REVIEW

File: dask/dataframe/io/tests/test_parquet.py [hunk 1]
```diff
@@ -2774,3 +2774,66 @@ def test_divisions_with_null_partition(tmpdir, engine):
 
     ddf_read = dd.read_parquet(str(tmpdir), engine=engine, index="a")
     assert ddf_read.divisions == (None, None, None)
+
+
+def test_parquet_pyarrow_write_empty_metadata(tmpdir):
+    # https://github.com/dask/dask/issues/6600
+    dd = pytest.importorskip("dask.dataframe")
```
Comment (rjzamora) [inline dask/dataframe/io/tests/test_parquet.py:2781]: I believe dask.dataframe is already imported as `dd` in the header, so you shouldn't need this in either of the new tests.

Comment (callumanoble) [reply dask/dataframe/io/tests/test_parquet.py:2781]: Thanks, this was lingering after move from a different file (where reqd.)

File: dask/dataframe/io/tests/test_parquet.py [hunk 2]
```diff
@@ -2774,3 +2774,70 @@ def test_divisions_with_null_partition(tmpdir, engine):
...
+        {"x": [1, 2, 1, 2], "y": [2, 0, 2, 0]}, dtype=("int64", "int64")
+    )
+
+    df1 = dd.from_delayed([df_a, df_b])
+    df1.to_parquet(
+        tmpdir,
+        engine="pyarrow",
+        partition_on=["x"],
+        append=False,
+        compute_kwargs={"scheduler": "synchronous"},
```
Comment (rjzamora) [inline dask/dataframe/io/tests/test_parquet.py:2825]: This is just for debugging convenience, correct? Can we remove for the test?

File: dask/dataframe/io/tests/test_parquet.py [hunk 3]
```diff
@@ -2774,3 +2774,70 @@ def test_divisions_with_null_partition(tmpdir, engine):
...
+
+    df = dd.from_delayed([df_a, df_b, df_c])
+
+    try:
+        df.to_parquet(
+            tmpdir,
+            engine="pyarrow",
+            partition_on=["x"],
+            append=False,
+            compute_kwargs={"scheduler": "synchronous"},
```
Comment (rjzamora) [inline dask/dataframe/io/tests/test_parquet.py:2801]: Same as below.

File: dask/dataframe/io/tests/test_parquet.py [hunk 4]
```diff
@@ -2774,3 +2774,70 @@ def test_divisions_with_null_partition(tmpdir, engine):
...
+    )
+
+    df2 = dd.from_delayed([df_c, df_d])
+    df2.to_parquet(
+        tmpdir,
+        engine="pyarrow",
+        partition_on=["x"],
+        append=True,
+        ignore_divisions=True,
+        compute_kwargs={"scheduler": "synchronous"},
```
Comment (rjzamora) [inline dask/dataframe/io/tests/test_parquet.py:2842]: Same as above.

**Review by rjzamora — APPROVED** [review]
Thanks for this fix @mcguipat - My only suggestion is to allow the default scheduler in the tests (but that is not critical).

File: dask/dataframe/io/tests/test_parquet.py (hunk 2 above)
Comment (callumanoble) [reply dask/dataframe/io/tests/test_parquet.py]: Correct, removed and pushed. Thanks

**Comment by martindurant** [comment]
Thanks for the review @rjzamora

**Pull request merged**